### ISSUE TITLE
add proxygen to osquery third-party

### ISSUE THREAD

**Issue by marpaia** [facebook/osquery#626]
For distributed queries

**Comment by marpaia** [comment]
Proxygen requires fbthrift and folly. This might not be as straightforward as we thought.

**Comment by marpaia** [comment]
I spent a bit of time working on this and chatting with @theopolis as well. Ultimately, we decided that the open source build support for proxygen isn't mature enough for us to effectively use it in osquery for a few reasons:

- Proxygen requires folly. Folly, although it's an awesome library, has non-existent build support for OS X, as well as lackluster build support for Linux. We have never used folly in osquery because of how poor the build system is, coupled with how many operating systems we need to support building on.
- Proxygen requires fbthrift. osquery uses Apache Thrift because it's more standard and there are way more existing packages and distributions for apache thrift. Using proxygen would require code-moding proxygen to support both, or refactoring osquery to use fbthrift (which would be unreasonable, given the lack of existing fbthrift packages).
- Proxygen doesn't offer any sustainable dependency resolution. There is only a shell script which will apt-get install dependencies. There's no CentOS dependency support, no OS X dependency support, no third-party vendoring system to allow someone to build static libraries completely from source, etc.

Given that build reliability is a top priority for osquery, if proxygen was not a Facebook library, we wouldn't think twice about using it. The juice doesn't seem worth the squeeze. Let's punt on using proxygen until the build/dependency support improves (or we get a chance to work with them to improve it).
